# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/xpconnect/tests/mochitest/test_bug916945.html
# Commit: 656b57e594de
# Full Hash: 656b57e594def0996f3f5518f36c6919636d9209
# Author: Olli Pettay <Olli.Pettay@helsinki.fi>
# Date: 2020-02-07 21:53:54
# Regressor Bug: 1578628
# File Overlap Count: 2
# Description:
#   Bug 1612431, it is enough to update existing browsing context name when changing iframe.name, r=kmag
#   
#   This brings back some of the behavior we had before bug 1578628.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D61637
# ==============================================================================

diff -r 60f14bea15af -r 656b57e594de js/xpconnect/tests/mochitest/test_bug916945.html
--- a/js/xpconnect/tests/mochitest/test_bug916945.html	Fri Feb 07 14:15:33 2020 +0100
+++ b/js/xpconnect/tests/mochitest/test_bug916945.html	Thu Feb 06 18:44:05 2020 +0000
@@ -28,6 +28,11 @@
 
     // Setting the 'name' attribute should propagate to the docshell.
     var ifrB = document.getElementById('ifrB');
+    await SpecialPowers.spawn(ifrB, [], () => {
+      Assert.equal(this.content.name, 'winB',
+                   'initial attribute value propagates to the docshell');
+    });
+
     ifrB.setAttribute('name', 'foo');
     await SpecialPowers.spawn(ifrB, [], () => {
       Assert.equal(this.content.name, 'foo',
@@ -46,6 +51,14 @@
     ok(!('foo' in window), "frame element name not resolved if it doesn't match the docshell");
     ok(!('bar' in window), "docshell name not resolved if it doesn't match the frame element");
 
+    // This shouldn't assert.
+    function listener() {
+      document.body.appendChild(ifrA);
+      ifrA.name = "";
+    }
+    ifrA.addEventListener("DOMNodeInserted", listener, {once: true});
+    document.body.appendChild(ifrA);
+
     SimpleTest.finish();
   }
 
